components/icons/lazyIcon.js

```diff
@@ -0,0 +1,225 @@
+import { oneOf } from 'prop-types'
+import { lazy, Suspense } from 'react'
+import { commonIconPropTypes } from './propTypes'
+
+export const lazyIcons = {
```

Comment:
@mohsenParvar @MortezaT I thought it's going to be done without using any mapper. is there any way to get the icon dynamically by iterating through all the icons in a specific directory?
like `return lazy(() => import(`./${name}Icon`).then(_ => ({ default: _.IconComponent })))`
or something similar?

Reply:
@majid-amiri 
We can use the name we receive from props for import and guess exported name and [errorBoundary](https://reactjs.org/docs/concurrent-mode-suspense.html#handling-errors) for error handling.
However what we did will detect it earlier and won't require `error boundary` for import itself.
I assume in our scenario we may need error boundary as well for rare cases. On the other hand we need to make sure added icon will be added to the list as well.
I'll change it to see if there would be any problems and let you know.

Reply:
@MortezaT ok good point. let's proceed with current solution to unblock dev guys, then we can improve this a lot in the next iteration.